compiler/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -445,8 +445,22 @@ impl FunctionContext<'_> {
         let type_size = Self::convert_type(element_type).size_of_type();
         let type_size =
             self.builder.numeric_constant(type_size as u128, NumericType::length_type());
-        // This shouldn't overflow as we are reaching for an initial array offset
-        // (otherwise it would have overflowed when creating the array)
+
+        let array_type = &self.builder.type_of_value(array);
+
+        // Checks for index Out-of-bounds
```

Comment:
Nice, I guess we are generating a bunch of repeat access checks for slices at the moment